src/modules/navigator/precland.cpp

```diff
@@ -493,7 +493,7 @@ bool PrecLand::check_state_conditions(PrecLandState state)
 		// if we're already in this state, only want to make it invalid if we reached the target but can't see it anymore
 		if (_state == PrecLandState::HorizontalApproach) {
 			if (fabsf(_target_pose.x_abs - vehicle_local_position->x) < _param_hacc_rad.get()
-			    && fabsf(_target_pose.y_rel - vehicle_local_position->y) < _param_hacc_rad.get()) {
+			    && fabsf(_target_pose.y_abs - vehicle_local_position->y) < _param_hacc_rad.get()) {
```

Comment:
This is a bug. But the fix should be in the other direction, see my other comment.

Reply:
@ndepal could you submit the correct fix as a separate PR? We are doing preclands and have been observing weird horizontal approach failures. Probably related to this.

Reply:
OK, I can do it.

@ndepal, you mean we just have to use `_target_pose.x_rel` instead of `_target_pose.x_abs - vehicle_local_position->x`?

Also, I wanted to notice that, this is incorrect calculation if we're inside the desired radius. The correct would be `sqrt(pow(_target_pose.x_rel, 2) + pow(_target_pose.y_rel, 2))`.

Reply:
@mhkabir Sure thing. But if @okalachev does it, that's great.

> you mean we just have to use _target_pose.x_rel instead of _target_pose.x_abs - vehicle_local_position->x?

Correct.

>The correct would be sqrt(pow(_target_pose.x_rel, 2) + pow(_target_pose.y_rel, 2)).

That would be the L2 norm. I used the L1 norm because it's cheaper to compute and just as good in this situation.

Reply:
@okalachev Is this addressed now?